zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Notebook.java

```diff
@@ -239,6 +239,8 @@ public Note cloneNote(String sourceNoteID, String newNoteName, AuthenticationInf
     Note newNote = createNote(subject);
     if (newNoteName != null) {
       newNote.setName(newNoteName);
+    } else {
+      newNote.setName("Note " + newNote.getId());
```

Comment:
@kavinkumarks `note.id()` is used widely and it's better to handle with another PR.